File: src/main/java/com/gitlab/daring/image/command/structure/MarkerType.java
package com.gitlab.daring.image.command.structure;

import com.gitlab.daring.image.command.parameter.IntParam;
import org.bytedeco.javacpp.opencv_core.Mat;
import org.bytedeco.javacpp.opencv_core.Point;
import org.bytedeco.javacpp.opencv_core.Scalar;

import java.awt.*;

import static com.gitlab.daring.image.util.GeometryUtils.getCenterRect;
import static com.gitlab.daring.image.util.OpencvConverters.toJava;
import static com.gitlab.daring.image.util.OpencvConverters.toOpencv;
import static org.bytedeco.javacpp.opencv_core.LINE_8;
import static org.bytedeco.javacpp.opencv_imgproc.circle;
import static org.bytedeco.javacpp.opencv_imgproc.rectangle;

public enum MarkerType {

    Rectangle, Circle;

    public void drawCenter(Mat m, IntParam p, int color, int th) {
        Dimension d = toJava(m.size());
        Rectangle cr = getCenterRect(d, p.getPv());
        Scalar c = Scalar.all(color);
        if (this == Rectangle) {
            rectangle(m, toOpencv(cr), c, th, LINE_8, 0);
        } else if (this == Circle) {
            Point cp = new Point(d.width / 2, d.height / 2);
            int radius = Math.min(cr.width, cr.height) / 2;
            circle(m, cp, radius, c, th, LINE_8, 0);
        } else {
            throw new IllegalArgumentException("markerType=" + this);
        }
    }
}

File: src/main/kotlin/com/gitlab/daring/image/command/structure/MarkerType.kt
package com.gitlab.daring.image.command.structure

import com.gitlab.daring.image.command.parameter.IntParam
import com.gitlab.daring.image.opencv.rect
import com.gitlab.daring.image.opencv.sizeDim
import com.gitlab.daring.image.util.GeometryUtils.getCenterRect
import org.bytedeco.javacpp.opencv_core.*
import org.bytedeco.javacpp.opencv_imgproc.circle
import org.bytedeco.javacpp.opencv_imgproc.rectangle

internal enum class MarkerType {

    Rectangle, Circle;

    fun drawCenter(m: Mat, p: IntParam, color: Int, th: Int) {
        val d = m.sizeDim
        val cr = getCenterRect(d, p.pv)
        val c = Scalar.all(color.toDouble())
        if (this == Rectangle) {
            rectangle(m, cr.rect, c, th, LINE_8, 0)
        } else if (this == Circle) {
            val cp = Point(d.width / 2, d.height / 2)
            val radius = Math.min(cr.width, cr.height) / 2
            circle(m, cp, radius, c, th, LINE_8, 0)
        } else {
            throw IllegalArgumentException("markerType=" + this)
        }
    }

}